Style Guide and Diataxis Review: Streams I (#227)
iotaledger/streams: documentation/docs/overview.md

- Application instances (channel identifier).
- Public keys of the publisher.
- Previous message ids (the message being linked to).
- Branch numbers (identifier for the specific branch).
- Sequencing numbers (the sequencing position of the publisher).

As messages are posted to and read from the channel, a local state for the user implementation will update with the message identifier, branch, and sequencing numbers for each publishing party. This allows user implementations to derive and search for the next message in the sequence to keep in sync. 

#### Single Branch Sequencing

In a single branch implementation, sequencing the state of each user will be updated to the same state. This means that regardless of the publisher, the state of each user will update the previous message id to the new message link, and the sequencing number will be incremented by one. 

Sequence states before Msg1 is sent:

> ```

Reviewer: ````suggestion

```plaintext
````

:::note

> We recommend that a single branch be used with only a single publisher to avoid out of sync parties from 
> issuing and retrieving messages from an incorrect sequencing combination (that is, Sub1 sends a message linked to Msg1 and Sub2 
> sends a message linked to that same Msg1 before seeing and registering the message from Sub1, thus forking the stream).

Reviewer: ```suggestion
We recommend that you use a single branch with only a single publisher to avoid out of sync parties from 
issuing and retrieving messages from an incorrect sequencing combination (that is, Sub1 sends a message linked to Msg1 and Sub2 
sends a message linked to that same Msg1 before seeing and registering the message from Sub1, thus forking the stream).
```

Multiple publishers can be used but it is important to ensure that they do not try to send messages in parallel to 
avoid this conflict. In the future, this may be an enforced limitation on single branch implementations to remove the 
possibility altogether. 

:::

#### Multi-Branch Sequencing 

> In a multi-branch implementation, the sequencing state of each user will be updated independently after each message is sent. To track the linking of messages of individual publishers within a tree-like structure, a secondary message is sent in tandem with every data message. This message is called a sequencing message and it contains the necessary essence for a user to derive the correct message id of a sequenced message. The sequencing messages are issued to an anchored branch generated during the creation of the channel. As a new message is generated by a user, a sequencing message is issued to this anchored branch to give users a reference guide to the location of the issued data packet.

Reviewer: There's a lot of passive voice in this paragraph

Author: 👍
